lib/srv/reexec.go

```diff
@@ -761,42 +662,223 @@ func runForward(handler forwardHandler) (errw io.Writer, code int, err error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	terminateChan := make(chan error, 1)
+	go func() {
+		buf := make([]byte, 1)
+		// Wait for the terminate file descriptor to be closed. The FD will be closed when Teleport
+		// parent process wants to terminate the remote command and all childs.
+		_, err := termiantefd.Read(buf)
```

Comment:
What's this (misspelled! 😬) pipe for? Can't a "graceful" shutdown just close the connection? (an ungraceful shutdown can just terminate the child I suppose)